### PR TITLE
[Ridesharing] fix greenlet_pool_for_ridesharing_services

### PR DESCRIPTION
**Pull request by xlqian**
when `greenlet_pool_for_ridesharing_services` is activated, the code crashed because of reading db inside of a greenelt

### CODE REVIEW

**Comment by xlqian** [comment]
https://jira.kisio.org/browse/NAVITIAII-3261

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py [hunk 1]
```diff
@@ -182,18 +195,18 @@ def decorate_journeys_with_ridesharing_offers(self, response, request, instance)
                             period_extremity = PeriodExtremity(section.end_date_time, False)
                         else:  # ridesharing at the end, we search for solution starting after the end of the pt sections
                             period_extremity = PeriodExtremity(section.begin_date_time, True)
-
+                        instance_params = self.InstanceParams.make_params(instance)
```
Comment (benoit-bst) [inline source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service_manager.py:198]: you can probably call it here : https://github.com/CanalTP/navitia/pull/3335/files#diff-58a666992e45cd17e1e1da8366f23cff7f1d8cbb03b84102293225f827713b04R174
At the top of the function ? Like that `greenlet_pool_actived` will be filled by `self.InstanceParams.make_params(instance) Class`

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service.py [hunk 2]
```diff
@@ -106,25 +106,28 @@ def _call_service(self, params, headers={}):
...
-            journeys = self._request_journeys(from_coord, to_coord, period_extremity, instance, limit)
+            journeys = self._request_journeys(from_coord, to_coord, period_extremity, instance_params, limit)
             feed_publisher = self._get_feed_publisher()
 
             self.record_call('ok')
 
             return journeys, feed_publisher
         except RidesharingServiceError as e:
             self.record_call('failure', **e.get_params())
+            self.logger.exception('')
```
Comment (azime) [inline source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service.py:126]: It's necessary ?

File: source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py [hunk 3]
```diff
@@ -270,8 +270,8 @@ def _request_journeys(self, from_coord, to_coord, period_extremity, instance, li
             raise RidesharingServiceError('non 200 response', resp.status_code, resp.reason, resp.text)
 
         if resp:
-            r = self._make_response(resp.json(), from_coord, to_coord, instance)
-            self.record_additional_info('Received ridesharing offers', nb_ridesharing_offers=len(r))
+            r = self._make_response(resp.json(), from_coord, to_coord, instance_params)
+            self.record_additional_info('   Received ridesharing offers', nb_ridesharing_offers=len(r))
```
Comment (azime) [inline source/jormungandr/jormungandr/scenarios/ridesharing/instant_system.py]: Minor:
```suggestion
            self.record_additional_info('Received ridesharing offers', nb_ridesharing_offers=len(r))
```

**Review by azime — COMMENTED** [review]
You have to rename instance to instance_params in the other connectors?

File: source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service.py (hunk 2 above)
Comment (xlqian) [reply source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service.py:126]: yes, there are some exceptions that are not logged... (PS: `self.logger.exception` logs the stack of exception, even though the message is empty)

**Review by nonifier — COMMENTED** [review]
I don't see any test that covers your fix :(

**Comment by xlqian** [comment]
@nonifier i'm working on it

**Comment by xlqian** [comment]
@nonifier here your are: https://github.com/CanalTP/navitia/pull/3335/files#diff-2c0388e092ac823785bd6b615bd217b1d07987dfe5a22b71e836c8c3cd6db56cR125

File: source/jormungandr/tests/ridesharing_service_manager_tests.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+def mock_http_instant_system(monkeypatch):
+    monkeypatch.setattr(
+        'jormungandr.scenarios.ridesharing.instant_system.InstantSystem._call_service', mock_instant_system
+    )
+
+
+@pytest.fixture(scope="function", autouse=True)
+def mock_instance_ridesharing_greenlet_pool_size(monkeypatch):
+    monkeypatch.setattr(
+        'jormungandr.instance.Instance.ridesharing_greenlet_pool_size', property(fget=lambda i: True)
```
Comment (benoit-bst) [inline source/jormungandr/tests/ridesharing_service_manager_tests.py:114]: :+1: :heart:

**Review by nonifier — APPROVED** [review]

**Pull request merged**